src/consensus/aft/raft.h

```diff
@@ -758,6 +776,7 @@ namespace aft
             become_retired(index, kv::RetirementPhase::Signed);
           }
           committable_indices.push_back(index);
+          is_new_primary = false;
```

Comment:
That seems a little brittle, I *think* we can't get a batch with two signatures inside at the moment, but it would be cleaner to reset on `get_signature_disposition()` to make sure we only force the one signature?

Reply:
So I think `get_signature_disposition()` is too early, as it risks resetting but then not producing a signature because of a later early-out. But I agree this is a little late, and potentially means we produce many signatures. I'll add an explicit reset of this once we think we've actually emitted a signature.

Reply:
Added this inside `get_signable_txids`, to avoid taking the lock twice, with comments around the reset and calling point to make the expectations clear. Considered `get_signable_txids_and_reset_should_sign`, but that makes it a little _too_ obvious this is a weird thing to do. Staring again, this bool could exist entirely in `history.h`, avoiding locking concerns and localising its impact, but I think I prefer the explicit naming of this in-consensus enum approach.

Reply:
That seems like a good compromise, I agree.